server/routes/api.js

```diff
@@ -92,6 +80,11 @@ router.post('/api/*', async ctx => {
     Object.assign(chooseToken, res);
   }
 
+  // todo: redirect is no effect
+  if (!chooseToken.access_token) {
+    ctx.redirect('/login?url=' + endUrl);
```

Comment:
`ctx.redirect` will redirect current request,  meaning `/api/apps` => `/api/login`. You can use `ctx.throw(...)`

Reply:
ok, change to:  ctx.throw(401, 'Unauthorized: invalid access token');  but still not implement page redirection when not found token.